watcherx/file.go

```diff
@@ -0,0 +1,118 @@
+package watcherx
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+)
+
+func WatchFile(ctx context.Context, file string, c EventChannel) error {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	dir := path.Dir(file)
+	if err := w.Add(dir); err != nil {
+		return errors.WithStack(err)
+	}
+	realFile, err := filepath.EvalSymlinks(file)
+	if err != nil {
+		if _, ok := err.(*os.PathError); ok {
+			// the file watcher should still watch the directory to get notified for
+			// file creation
+			realFile = ""
+		} else {
+			return errors.WithStack(err)
+		}
+	}
+	if realFile != "" && realFile != file {
+		// file is a symlink, we have to explicitly watch the referenced file.
+		// We are watching file instead of lastFile because fsnotify identifies
+		// watch entries by the passed name but follows symlinks when watching
+		// (at least on unix but not on windows).
```

Comment:
I think this needs clarification (please read and correct any misunderstanding):

```suggestion
		// If `realFile` != `file` then `file` is a symlink and we have to explicitly watch the referenced file.
		// This is because fsnotify follows symlinks and watches the destination file, not the symlink
		// itself. That is at least the case for unix systems. See: https://github.com/fsnotify/fsnotify/issues/199
```